### PR TITLE
Pattern matching and utilization of the new switch syntax for the java16 branch

### PR DESCRIPTION
**Pull request by md5sha256**
## Proposed changes

- Use the new pattern matching (instanceof) syntax in a few places
- Use the new switch syntax in a few places

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/WoodcutterAndroid.java [hunk 1]
```diff
@@ -92,70 +92,46 @@ private void replant(@Nonnull Block block) {
         Predicate<Material> soilRequirement = null;
 
         switch (logType) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/WoodcutterAndroid.java:94]:
```suggestion

        // TODO: Create a record that holds `saplingType` and `soilRequirement` to further simplify this.
        switch (logType) {
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java [hunk 2]
```diff
@@ -32,18 +32,12 @@ public HerculesPickaxe(ItemGroup itemGroup, SlimefunItemStack item, RecipeType r
...
-                            break;
-                        default:
-                            break;
+                        case DEEPSLATE_IRON_ORE -> drops.add(new CustomItemStack(SlimefunItems.IRON_DUST, 2));
+                        case DEEPSLATE_GOLD_ORE -> drops.add(new CustomItemStack(SlimefunItems.GOLD_DUST, 2));
+                        case COPPER_ORE, DEEPSLATE_COPPER_ORE ->
+                                drops.add(new CustomItemStack(SlimefunItems.COPPER_DUST, 2));
+                        default -> {
+                        }
                     }
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java:41]: There is another (almost equivalent) switch/case a few lines down.
Why was this updated and the one below not?

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
There is a switch/case you forgot in one of the files.

Also, could you please add that todo comment for me?
It should not be part of this PR but I don't wanna forget about it.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java (hunk 2 above)
Comment (md5sha256) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java:41]: Missed it when doing the auto-update using my IDE. Fixed now

**Comment by md5sha256** [comment]
Whoa, why did that happen? I was trying to fix the merge conflicts but I must have done something wrong

File: CHANGELOG.md [hunk 3]
```diff
@@ -41,7 +41,6 @@
 * (API) Added "WOOL_CARPETS" tag (for compatibility across MC 1.19/1.18 tags)
 
 #### Changes
-* Slimefun now requires Java 16
```
Comment (TheBusyBiscuit) [inline CHANGELOG.md:44]: Please don't remove this from the changelog :(

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BeeWingsListener.java [hunk 4]
```diff
@@ -36,12 +36,10 @@ public BeeWingsListener(@Nonnull Slimefun plugin, @Nonnull BeeWings wings) {
 
     @EventHandler(ignoreCancelled = true)
     public void onApproachGround(EntityToggleGlideEvent e) {
-
         if (wings == null || !e.isGliding() || wings.isDisabled() || !(e.getEntity() instanceof Player)) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BeeWingsListener.java:39]:
```suggestion
        if (wings == null || !e.isGliding() || wings.isDisabled() || !(e.getEntity() instanceof Player player)) {
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by TheBusyBiscuit** [comment]
Closing this for now, so that we can get #3602 rolled out asap

**Pull request closed without merging**